### PR TITLE
Use Swift Foundation Result type to represent a value and error in ex…

### PR DESCRIPTION
**Pull request by DJBen**
…ample project

### CODE REVIEW

**Comment by sebastianv1** [comment]
Love it! Do you mind adding this to the `CHANGELOG`

**Comment by DJBen** [comment]
> Love it! Do you mind adding this to the `CHANGELOG`

Will do!

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,3 +17,9 @@
 * Retain subcomponent graphs  
   [Sebastian Shanus](https://github.com/sebastianv1)
   [#42](https://github.com/square/Cleanse/issues/42)
+
+#### Improvements
+
+* Replace demo app's ErrorOptional type with Swift Foundation Result type
```
Comment (sebastianv1) [inline CHANGELOG.md]: Two spaces at the end of this line (creates a newline)

**Comment by sebastianv1** [comment]
Will merge and cleanup

**Pull request merged**